storage/engine/rocksdb/db.cc

```diff
@@ -39,33 +39,84 @@
 extern "C" {
 #include "_cgo_export.h"
 
-struct DBBatch {
+struct DBEngine {
+  rocksdb::DB* const rep;
+
+  DBEngine(rocksdb::DB* r)
+      : rep(r) {
+  }
+  virtual ~DBEngine() { }
+
+  virtual DBStatus Put(DBSlice key, DBSlice value);
+  virtual DBStatus Merge(DBSlice key, DBSlice value);
+  virtual DBStatus Delete(DBSlice key);
+  virtual DBStatus WriteBatch();
+  virtual DBStatus Get(DBSlice key, DBString* value);
+  virtual DBIterator* NewIter();
+};
+
+struct DBImpl : public DBEngine {
```

Comment:
s/DBImpl/DBOwner/, since the sole difference between this class and `DBEngine` is that it owns and deletes the underlying DB?